### ISSUE TITLE
Document backporting of code

### ISSUE THREAD

**Issue by LKreutzer** [magma/magma#14936]
Changes to release branches require some special treatment. It should be documented 

- How to backport code
  - Create PR on master - get merged
  - Add label to PR -> trigger backporting workflow.
  - New PR is created by [MagmaCIBot](https://github.com/MagmaCIBot).
  - Ask for approval in release Slack channel.
  - Example PR https://github.com/magma/magma/pull/14763 and backport PR https://github.com/magma/magma/pull/14945 
- Contacts for questions
- CI workflows for backports
  - `backport-pull-request.yml`
- :warning: It should be explained that documentation for the docusaurus is **never** backported, because the docusaurus is deployed from the versioned docs on the `master` branch, see `docs/docusaurus/versioned_docs`.
- `README.md` files can be backported if it makes sense.

**Comment by LKreutzer** [comment]
I would add the following text to a new section on the https://github.com/magma/magma/wiki/Contributing-Code page.

## Back-porting changes to release branches

The process for back-porting code changes to release branches, such as `v1.8`, is as follows:

- Create and merge a pull-request to the `master` branch.
- On the merged pull-request, add the label `apply<branch_name>`, e.g. `apply-v1.8`. This change triggers the back-porting workflow, see `backport-pull-request.yml`.
- A new (back-port) pull-request is automatically created by the [MagmaCIBot](https://github.com/MagmaCIBot).
- This new pull-request should be reviewed and posted in the [#magma-releases](https://magmacore.slack.com/archives/C0301NRRL7K) Slack channel, where the managers of the release need to approve the change for back-porting.
- Once the pull-request is approved it can be merged.

For example, the pull-request https://github.com/magma/magma/pull/14763 was merged to the master branch. The label `apply-v1.8` was added and the back-port pull-request https://github.com/magma/magma/pull/14945 was created and merged.

:warning: The above steps apply to code changes that are merged to the `master` branch as well. If a change is only needed for a release branch, but should not be applied to the `master` branch, then a regular pull-request on the release branch should be opened and announced in the [#magma-releases](https://magmacore.slack.com/archives/C0301NRRL7K) Slack channel.

:warning: Changes to the documentation in the docusaurus is **never** backported, because the docusaurus is deployed from the versioned docs on the `master` branch, see `docs/docusaurus/versioned_docs` and the [contributing documentation page](https://github.com/magma/magma/wiki/Contributing-Documentation#modifying-the-documentation). Documentation in `README.md` outside of the docusaurus can be back-ported if necessary.

**Comment by LKreutzer** [comment]
@voisey please take a look at the above draft and let me know if there is something to improve.

**Comment by voisey** [comment]
Looks good! A few small questions/comments:

- Is there a reason you use "back-porting" rather than "backporting"? I think I've only ever seen the latter and it seems to be more common when I search for it, e.g. it's what's used on [Wikipedia](https://en.wikipedia.org/wiki/Backporting). Similar question with "pull-request" vs "pull request".
- I don't quite understand the opening sentence of the first warning, i.e. "The above steps apply..." What's the difference between this sentence and what's being covered above?
- "Changes to the documentation in the docusaurus is..." -> "Changes to the documentation in the docusaurus are..."
- "Documentation in README.md outside" -> "Documentation in README.md files outside"

**Comment by LKreutzer** [comment]
Addressed comments and published the section.

**Comment by LKreutzer** [comment]
Announced in Slack

- https://magmacore.slack.com/archives/C018J8UMGMR/p1675693956829549
- https://magmacore.slack.com/archives/C0301NRRL7K/p1675693984180299
